lib/ansible/module_utils/network/skydive/api.py

```diff
@@ -0,0 +1,162 @@
+        kwargs['scheme'] = "http"
+        if 'ssl' in kwargs:
+            if kwargs['ssl']:
+                kwargs['scheme'] = "https"
+        if 'insecure' not in kwargs:
+            kwargs['insecure'] = False
+        self.restclient_object = RESTClient(kwargs['endpoint'],
+                                            scheme=kwargs['scheme'],
+                                            insecure=kwargs['insecure'],
+                                            username=kwargs['username'],
+                                            password=kwargs['password'])
+
+
+class skydive_lookup(skydive_restclient):
+    provider_spec = {'provider': dict(type='dict', options=SKYDIVE_PROVIDER_SPEC)}
+
+    def __init__(self, provider):
+        super(skydive_lookup, self).__init__(**provider)
+        self.query_str = ""
+
+    def lookup_query(self, filter_data):
+        query_key = filter_data.keys()[0]
+        self.query_str = filter_data[query_key]
+        nodes = self.restclient_object.lookup_nodes(self.query_str)
+        return nodes[0].__dict__
+
+
+class skydive_flow_topology(skydive_restclient):
+    ''' Implements Skydive Flow capture modules '''
+    def __init__(self, module):
+        self.module = module
+        provider = module.params['provider']
+
+        super(skydive_flow_topology, self).__init__(**provider)
+
+    def run(self, ib_spec):
+        state = self.module.params['state']
+        if state not in ('present', 'absent'):
+            self.module.fail_json(msg='state must be one of `present`, `absent`, got `%s`' % state)
+
+        result = {'changed': False}
+        obj_filter = dict([(k, self.module.params[k]) for k, v in iteritems(ib_spec) if v.get('ib_req')])
+
+        proposed_object = {}
+        for key in iterkeys(ib_spec):
+            if self.module.params[key] is not None:
+                proposed_object[key] = self.module.params[key]
+
+        capture_str = SKYDIVE_GREMLIN_QUERY + "('Name', '{0}', 'Type', '{1}', 'extra_tcp_metric', '{2}'\
+                                                 ,'ip_defrag', '{3}', 'reassemble_tcp',\
+                                                 '{4}', 'layer_key_mode', '{5}')".format(obj_filter['name'],
+                                                                                         obj_filter['capture_type'],
+                                                                                         obj_filter['extra_tcp_metric'],
+                                                                                         obj_filter['ip_defrag'],
+                                                                                         obj_filter['reassemble_tcp'],
+                                                                                         obj_filter['layer_key_mode'])
+        # to check current object ref for idempotency
+        captured_list_objs = self.restclient_object.capture_list()
+        current_ref = False
```

Comment:
Why not setting it to None and directly capturing the UUID. This would spare you the second pass for the delete case would not cost much in the creation case. You could/should also return this uuid as part of the result. As shown by the API, uuid are more trusted than names and users should be able to delete captures based on the uuid. In fact names are not mandatory.